### PR TITLE
OCPBUGS-10821: Update ACME solver image to read from env 

### PR DESCRIPTION
**Pull request by thejasn**
Description
---

Adds a new ENV variable to read the ACME solver image at runtime. Related PR on midstream will update the image to reference Red Hat's image. 

Related Midstream PR: https://gitlab.cee.redhat.com/cpaas-midstream-cfe/openshift-cert-manager/-/merge_requests/96

### CODE REVIEW

File: config/manager/manager.yaml [hunk 1]
```diff
@@ -39,47 +39,49 @@ spec:
...
-        securityContext:
-          allowPrivilegeEscalation: false
-          capabilities:
-            drop:
-              - "ALL"
-          privileged: false
-          runAsNonRoot: true
-          seccompProfile:
-            type: "RuntimeDefault"
+        - command:
```
Comment (swghosh) [inline config/manager/manager.yaml:42]: This diff seems to be larger than usual, not sure why. Though you'd only change a couple of lines and the whole block as-is.

File: bundle/manifests/operator.openshift.io_certmanagers.yaml [hunk 2]
```diff
@@ -290,6 +294,10 @@ spec:
                       - name
                       type: object
                     type: array
+                  overrideLabels:
```
[the remark this answers is not in the document]
Comment (thejasn) [reply bundle/manifests/operator.openshift.io_certmanagers.yaml:297]: Can you check on `master` by running `make bundle`?

File: config/manager/manager.yaml (hunk 1 above)
Comment (thejasn) [reply config/manager/manager.yaml:42]: Seems that the indentation was incorrect

Comment (swghosh) [reply config/manager/manager.yaml:42]: oh cool

File: bundle/manifests/operator.openshift.io_certmanagers.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (swghosh) [reply bundle/manifests/operator.openshift.io_certmanagers.yaml:297]:
```
$ make bundle
go run sigs.k8s.io/controller-tools/cmd/controller-gen rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
/home/swghosh/Desktop/repositories/cert-manager/cert-manager-operator/bin/operator-sdk generate kustomize manifests -q
cd config/manager && go run sigs.k8s.io/kustomize/kustomize/v4 edit set image controller=openshift.io/cert-manager-operator:
go run sigs.k8s.io/kustomize/kustomize/v4 build config/manifests | /home/swghosh/Desktop/repositories/cert-manager/cert-manager-operator/bin/operator-sdk generate bundle -q --version 1.11.0 --channels="stable-v1,stable-v1.11" --default-channel=stable-v1
INFO[0001] Creating bundle.Dockerfile                   
INFO[0001] Creating bundle/metadata/annotations.yaml    
INFO[0001] Bundle metadata generated successfully       
/home/swghosh/Desktop/repositories/cert-manager/cert-manager-operator/bin/operator-sdk bundle validate bundle
INFO[0000] All validation tests have completed successfully

$ git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   bundle/manifests/operator.openshift.io_certmanagers.yaml

no changes added to commit (use "git add" and/or "git commit -a")

$ git diff
diff --git a/bundle/manifests/operator.openshift.io_certmanagers.yaml b/bundle/manifests/operator.openshift.io_certmanagers.yaml
index f4911dee..08feb9ab 100644
--- a/bundle/manifests/operator.openshift.io_certmanagers.yaml
+++ b/bundle/manifests/operator.openshift.io_certmanagers.yaml
@@ -160,6 +160,10 @@ spec:
                       - name
                       type: object
                     type: array
+                  overrideLabels:
+                    additionalProperties:
+                      type: string
+                    type: object
                 type: object
               controllerConfig:
                 description: "ControllerConfig specifies further customization options
@@ -290,6 +294,10 @@ spec:
                       - name
                       type: object
                     type: array
+                  overrideLabels:
+                    additionalProperties:
+                      type: string
+                    type: object
                 type: object
               logLevel:
                 default: Normal
@@ -470,6 +478,10 @@ spec:
                       - name
                       type: object
                     type: array
+                  overrideLabels:
+                    additionalProperties:
+                      type: string
+                    type: object
                 type: object
             type: object
           status:
```

Yeah, it is missing in current master. I apprehend it to be lost between merge conflict of #113 + #112, my bad.

**Comment by xingxingxia** [comment]
Adding label for merge. Will choose to do post-merge test for the my bug per other on-hand work items.
/label qe-approved

**Comment by swghosh** [comment]
/cc @xenolinux 
for docs-approved label
/cc @davemulford 
for px-approved label

**Comment by thejasn** [comment]
/label docs-approved
re-applying due to bot issues

**Pull request merged**